Bicep.sln

```diff
@@ -88,16 +88,14 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bicep.Local.Extension", "sr
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bicep.Local.Deploy", "src\Bicep.Local.Deploy\Bicep.Local.Deploy.csproj", "{3F3A6387-DD5F-40A6-89D1-92653E4872D3}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bicep.Local.Deploy.IntegrationTests", "src\Bicep.Local.Deploy.IntegrationTests\Bicep.Local.Deploy.IntegrationTests.csproj", "{3F25D072-7A7E-419D-8425-6F2C7CF42BFD}"
-EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bicep.Local.Extension.Mock", "src\Bicep.Local.Extension.Mock\Bicep.Local.Extension.Mock.csproj", "{A7D359D9-654A-4FAF-9BC2-DA9667EF8756}"
-EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "IO", "IO", "{83EF0BE8-E57C-499D-ABC8-019A7F68F407}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Bicep.IO", "src\Bicep.IO\Bicep.IO.csproj", "{A8392FEA-CC1F-4630-85A0-E6BA97AD87D7}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Bicep.IO.UnitTests", "src\Bicep.IO.UnitTests\Bicep.IO.UnitTests.csproj", "{69FC508E-6FE7-4C01-942F-F3BF0F3C68E3}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Bicep.Local.Extension.UnitTests", "src\Bicep.Local.Extension.UnitTests\Bicep.Local.Extension.UnitTests.csproj", "{FBCC18D4-4FBC-456C-8295-F2AF5ABB42B7}"

```

Comment:
There's a lot going on in this PR, but I like the direction it is taking. I think it would be simpler to review if we could break it up into pieces, and handle this in a bit more of an incremental approach.

Some ideas on sequencing:
1. Add in reflection-based type generation capabilities with the `--describe` flag
1. Add in builder-style APIs for simplifying CLI boilerplate
1. Add in strongly-typed handlers (ideally in a back-compatible manner, so that there's an easy upgrade path)